File: client_libraries/nodejs_library/smoke-test/street_view_publish_service_smoke_test.js
// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

'use strict';

const assert = require('assert');

describe('StreetViewPublishServiceSmokeTest', () => {
  it('should return INVALID_ARGUMENT', done => {
    const google = require('../src');

    const client = new google.v1.StreetViewPublishServiceClient({});

    const photo = {};
    client.createPhoto(photo, (err, photo) => {
        assert(err instanceof Error);
        assert.strictEqual(err.code, 3);
        done();
      });
  });
});

File: client_libraries/java_library/java-streetview-publish/src/test/java/com/google/streetview/publish/v1/StreetViewPublishServiceSmokeTest.java
/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.google.streetview.publish.v1;

import com.google.api.gax.rpc.InvalidArgumentException;
import com.google.api.gax.rpc.StatusCode.Code;
import com.google.common.base.Preconditions;
import com.google.geo.ugc.streetview.publish.v1.StreetViewPublishResources.Photo;
import java.util.logging.Level;
import java.util.logging.Logger;
import org.junit.Test;

@javax.annotation.Generated("by GAPIC")
public class StreetViewPublishServiceSmokeTest {
  private static final String PROJECT_ENV_NAME = "GOOGLE_CLOUD_PROJECT";
  private static final String LEGACY_PROJECT_ENV_NAME = "GCLOUD_PROJECT";

  @Test
  public void run() {
    main(null);
  }

  public static void main(String args[]) {
    Logger.getLogger("").setLevel(Level.WARNING);
    try {
      execute();
      System.out.println("OK");
    } catch (Exception e) {
      System.err.println("Failed with exception:");
      e.printStackTrace(System.err);
      System.exit(1);
    }
  }

  public static void execute() throws Exception {
    try (StreetViewPublishServiceClient client = StreetViewPublishServiceClient.create()) {
      Photo photo = Photo.newBuilder().build();

      Photo response = client.createPhoto(photo);
    } catch (InvalidArgumentException e) {
      // We expect createPhoto to return INVALID_ARGUEMENT because the photo is empty.
      if (e.getStatusCode().getCode() != Code.INVALID_ARGUMENT) {
        throw e;
      }
    }
  }

  private static String getProjectId() {
    String projectId = System.getProperty(PROJECT_ENV_NAME, System.getenv(PROJECT_ENV_NAME));
    if (projectId == null) {
      projectId =
          System.getProperty(LEGACY_PROJECT_ENV_NAME, System.getenv(LEGACY_PROJECT_ENV_NAME));
    }
    Preconditions.checkArgument(projectId != null, "A project ID is required.");
    return projectId;
  }
}
